apiserver/facades/client/application/deploy.go

```diff
@@ -184,10 +185,21 @@ func DeployApplication(
 			}
 		}
 
+		charmResources := args.Charm.Meta().Resources
+		resolvedResources := make(applicationservice.ResolvedResources, 0,
+			len(charmResources))
+		for _, res := range charmResources {
+			resolvedResources = append(resolvedResources, applicationservice.ResolvedResource{
+				Name:   res.Name,
+				Origin: charmresource.OriginUpload,
```

Comment:
I can't work out why we would want these all to be origin upload, is there something I've missed? 

This method is called for both charmhub charms and for local charms [here](https://github.com/juju/juju/blob/685c05c460703aee657482c59595a93a4391612c/apiserver/facades/client/application/application.go#L543), at least that is what the comment says. If it is the case that in 4.0 that we are explicitly only using this method for local charms this should be updated and there should be a check to prevent non local charms being deployed this way.